exercises/rectangles/example.lua

```diff
@@ -0,0 +1,94 @@
+local combinations
+
+local function generate_combinations(a, n, offset)
+  if n > (#a - offset) then
+    return
+  end
+
+  if n == 0 then
+    coroutine.yield({})
+    return
+  end
+
+  for v in combinations(a, n - 1, offset + 1) do
+    table.insert(v, 1, a[1 + offset])
+    coroutine.yield(v)
+  end
+
+  for v in combinations(a, n, offset + 1) do
+    coroutine.yield(v)
+  end
+end
+
+function combinations(a, n, offset)
+  return coroutine.wrap(function() generate_combinations(a, n, offset or 0) end)
+end
```

Comment:
This is a little snippet that I've used a couple of times for generating combinations. I reproduced it here without changes instead of writing it again.